### PR TITLE
Fix/create language context

### LINKED ISSUE #31: I18n Language Change Props Not Propagating
Our vision is to support multiple languages with the ParetOS. To do so, we have added the Amplify I18N library to help us with replacing certain UI/UX language based on the device's detected language. Right now, we support Spanish and Luganda. 

However, there is a problem. In `App.js` there is a language bar, in the side-nav, where someone can change the active language of the application. When they change the language, they can see the language changing in the side-nav - but not in the other components, until they migrate to another container (using the `history` object). Then, the language propagates. My theory is that these language props are not being propagated correctly somehow, perhaps the event needs to be triggered outside the side-nav? Maybe do some research on Google.

### PR DESCRIPTION
**Pull request by jayeclark**
Pull-Request for `paretOS`

## Description
Adds language context and creates a 'single source of truth' for available languages and updating the Language on the front end and back end. (Functions & UX are now standardized across the LeftNav dropdown and the EditProfile LanguageSelector.)

(It ended up not being necessary to force re-rendering of the components.)

## Relates to
Fixes #31 
There are a fair number of translations that are currently missing/not passing through I18n or not yet translated - so if you're testing locally to validate that this fix works, try to navigate to several different areas of the app (you may be on a view where most translations are missing.) I'm going to be opening some issues to clean up the translations.

## Reviewers
- @mikhael28 (merge duty)

### Screenshots / other info
lang

### CODE REVIEW

**Review by mikhael28 — APPROVED** [review]
Great work! I left one little comment about removing what seems to be an unnecessary div/replacing it with a <> at the least. Once you do that, please go ahead and merge!

File: src/profile/LanguageSelector.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  const handleChange = (e) => {
+    updateLanguage({
+      language: availableLanguages.find((x) => x.code === e.target.value),
+      id: props.id,
+      setLanguage: handleSetLanguage,
+      setIsLoading: handleSetIsLoading,
+    });
+  };
+  return (
+    <div>
```
[the remark this answers is not in the document]
Comment (jayeclark) [reply src/profile/LanguageSelector.js:30]: Yep, of course!

```
<div>
  <div>
    <div>
      <div>
        <div>
          <div>
            <div>
              <div>
                <div>
                  <div>
                    <div>
                      <div>
                        Oops, old (bad) habits die hard!
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
```

File: src/index.css [hunk 2]
```diff
@@ -298,6 +298,7 @@ a.MuiButton-root:focus {
 
 .flex {
   display: flex;
+  align-items: center;
```
[the remark this answers is not in the document]
Comment (jayeclark) [reply src/index.css:301]: I haven't done an exhaustive review (figured I'd do that later and open up issues as I found anything amiss) but I looked through enough instances where flex was used that I feel comfortable that adding it it isn't going to horrifically break anything (and might unexpectedly improve some alignments as well.)

[the remark this answers is not in the document]
Comment (mikhael28) [reply src/index.css:301]: Sounds great! Feel free to merge whenever.

**Pull request merged**